overlord/hookstate/hookmgr.go

```diff
@@ -396,10 +397,16 @@ func trackHookError(context *Context, output []byte, err error) {
 	if context.setup.IgnoreError {
 		extra["IgnoreError"] = "1"
 	}
-	oopsid, err := errtrackerReport(context.SnapName(), errmsg, dupSig, extra)
-	if err == nil {
-		logger.Noticef("Reported hook failure from %q for snap %q as %s", context.HookName(), context.SnapName(), oopsid)
-	} else {
-		logger.Debugf("Cannot report hook failure: %s", err)
+
+	context.state.Lock()
+	problemReportsDisabled := settings.ProblemReportsDisabled(context.state)
+	context.state.Unlock()
+	if !problemReportsDisabled {
```

Comment:
I know why you did it this way but it would be more natural if we could avoid the `!fooDisabled` pattern and instead use `fooEnabled`

Reply:
Maybe refactoring the code like this would make it a bit easier to follow:
```
if problemReportsDisabled {
    return
}

oopsid, err := errtrackerReport(context.SnapName(), errmsg, dupSig, extra)
...
```